hummingbot/connector/exchange/bitmax/bitmax_exchange.py

```diff
@@ -0,0 +1,961 @@
+            except asyncio.CancelledError:
+                raise
+            except Exception as e:
+                self.logger().error(str(e), exc_info=True)
+                self.logger().network("Unexpected error while fetching account updates.",
+                                      exc_info=True,
+                                      app_warning_msg="Could not fetch account updates from Bitmax. "
+                                                      "Check API key and network connection.")
+                await asyncio.sleep(0.5)
+
+    async def _update_balances(self):
+        """
+        Calls REST API to update total and available balances.
+        """
+        local_asset_names = set(self._account_balances.keys())
+        remote_asset_names = set()
+        response = await self._api_request("get", "cash/balance", {}, True, force_auth_path_url="balance")
+        for account in response["data"]:
+            asset_name = account["asset"]
+            self._account_available_balances[asset_name] = Decimal(account["availableBalance"])
+            self._account_balances[asset_name] = Decimal(account["totalBalance"])
+            remote_asset_names.add(asset_name)
+
+        asset_names_to_remove = local_asset_names.difference(remote_asset_names)
+        for asset_name in asset_names_to_remove:
+            del self._account_available_balances[asset_name]
+            del self._account_balances[asset_name]
+
+    async def _update_order_status(self):
+        """
+        Calls REST API to get status update for each in-flight order.
+        """
+        last_tick = int(self._last_poll_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL)
+        current_tick = int(self.current_timestamp / self.UPDATE_ORDER_STATUS_MIN_INTERVAL)
+
+        if current_tick > last_tick and len(self._in_flight_orders) > 0:
+            tracked_orders = list(self._in_flight_orders.values())
+            tasks = []
+            for tracked_order in tracked_orders:
+                order_id = await tracked_order.get_exchange_order_id()
+                tasks.append(self._api_request(
+                    "get",
+                    f"cash/order/status?orderId={order_id}",
+                    {},
+                    True,
+                    force_auth_path_url="order/status")
+                )
+            self.logger().debug(f"Polling for order status updates of {len(tasks)} orders.")
+            update_results = await safe_gather(*tasks, return_exceptions=True)
+            for update_result in update_results:
+                if isinstance(update_result, Exception):
+                    raise update_result
+                if "data" not in update_result:
+                    self.logger().info(f"_update_order_status result not in resp: {update_result}")
+                    continue
+                # for trade_msg in update_result["data"]["trade_list"]:
+                #     await self._process_trade_message(trade_msg)
```

Comment:
Might I ask about the decision to omit this?

Reply:
I couldn't find anywhere in the API that would give me _my_ trades, only public trades endpoint/subscription. Maybe am being dense though? 😆

- restful API orders: https://bitmax-exchange.github.io/bitmax-pro-api/#order
- WS API trades: https://bitmax-exchange.github.io/bitmax-pro-api/#channel-market-trades
- WS API orders: https://bitmax-exchange.github.io/bitmax-pro-api/#channel-order-and-balance